### ISSUE TITLE
profiler: pprof memory leak

### ISSUE THREAD

**Issue by Rossiar** [googleapis/google-cloud-go#1413]
## Client

Stackdriver Profiler

## Describe Your Environment

Alpine Docker on Kubernetes in GKE, my service is running 3 pods to a deployment with 50mb max memory per pod, I expect each pod to normally consume ~15-20mb. Over the course of ~6hrs the memory creeps up from this expected value to 50mb, where the pod is then killed by kubernetes. 

## Expected Behavior

The profiler does not cause a memory leak in my application, as we added the profiler to our code as an atomic change without modifying any other part of the codebase - I would not expect it to have an impact.

## Actual Behavior

The profiler has a memory leak that seems to originate somewhere in pprof:
![profile2](https://user-images.githubusercontent.com/1635940/56959667-18d4e900-6b46-11e9-8234-eb8ff6bafe96.png)

We have now seen this in 2 separate applications after importing the profiler, the image above showing a comparison of about 1hr after the app started to about 30min before it was killed.

**Comment by aalexand** [comment]
@Rossiar If you change the weight filter to show the top bucket of the heap profiles here, does it show that the profiler call path has more memory attributed to it?

In general, I wouldn't be too surprised if some call paths in the profiler would get some amount of memory in-use attributed since Go is a GC language and things can escape the function scope, but I would be surprised if this doesn't get timely garbage collected and causes an OOM.

**Comment by Rossiar** [comment]
@aalexand the profiler is taking a large share of memory, but to be honest the heap was not actually the culprit here since the entire heap is only 8mb at max anyway, so I'm really treating this figure as an indicator of potential trouble elsewhere in memory:
![profileTotal](https://user-images.githubusercontent.com/1635940/57010894-f9e25f80-6bf6-11e9-8734-ff9a85adcb21.png)

Here is the allocated heap for the same timeframe:
![profileTotal](https://user-images.githubusercontent.com/1635940/57011029-9dcc0b00-6bf7-11e9-92a8-c5a05f8125b0.png)

We also have some Prometheus monitoring in place that exposes some stats about our overall memory usage, you can see below when I deployed the monitoring and the app began leaking until OOM kill.
![image](https://user-images.githubusercontent.com/1635940/57010871-e7682600-6bf6-11e9-8bf7-4a1ecf2aee35.png)

What is even more interesting is that our staging environment (where traffic is much lighter) also experienced this issue (albeit at a slower rate of increase) when adding the profiler.

I think the next step is to redeploy my application into stage with the profiler and take a pprof heap dump when memory is well above requested, it will take a few hours to let the leak grow to the right size, so I will post here again when it is ready.

**Comment by Rossiar** [comment]
@aalexand unfortunately I am unable to recreate the issue on our staging environment despite using the same source code (I branched from the tag that I was seeing the leak with to add `pprof` and the issue disappeared in the new branch that I pushed). This leads me to believe that the issue may lie in our project dependencies - I do not have a whole lot of faith in `dep`  or its ability to correctly cache major vs minor versions in our CI - so perhaps there is something else to blame here.

I think it is best to close this issue as I don't see a clear way to continue investigating.
